### ISSUE TITLE
Argument error with standard_error and unicode with Erlang 18 

### ISSUE THREAD

**Issue by stavro** [elixir-lang/elixir#3594]
I've been receiving the following compilation error **periodically** (not every time) on SemaphoreCI.  I cannot reproduce the compilation error locally.  It's also not always the same file which throws the compilation error each time.

Have you seen this before and/or do you have any ideas on how I may fix it?

[Here's the file it choked on](https://github.com/stavro/arc/blob/master/lib/arc/definition.ex) for this example.

```
...
Compiled src/erlcloud_sdb.erl
Compiled src/erlcloud_ses.erl
Compiled src/erlcloud_sts.erl
src/erlcloud_util.erl:38: Warning: crypto:md5/1 is deprecated and will be removed in in a future release; use crypto:hash/2
Compiled src/erlcloud_util.erl
Compiled src/erlcloud_sns.erl
Compiled src/erlcloud_sqs.erl
==> arc
Compiled lib/arc.ex

== Compilation error on file lib/arc/definition.ex ==
** (ArgumentError) argument error
    (stdlib) :io.put_chars(:standard_error, :unicode, [['lib/arc/definition.ex', 58, '2', 58, 32], 'warning: ', [118, 97, 114, 105, 97, 98, 108, 101, 32, 'options', 32, 105, 115, 32, 117, 110, 117, 115, 101, 100], 10])
    (elixir) src/elixir_errors.erl:255: :elixir_errors.do_warn/1
    (stdlib) lists.erl:1337: :lists.foreach/2
    (stdlib) erl_eval.erl:669: :erl_eval.do_apply/6


05:15:33.205 [error] Process #PID<0.49.0> raised an exception
** (ArgumentError) argument error
    (stdlib) :io.put_chars(:standard_error, :unicode, ["** (ArgumentError) argument error\n    (stdlib) :io.put_chars(:standard_error, :unicode, [[[[[[] | \"\\e[31m\"] | \"\\e[1m\"], \"could not compile dependency arc, mix compile failed. You can recompile this dependency with `mix deps.compile arc` or update it with `mix deps.update arc`\"] | \"\\e[0m\"], 10])\n    (mix) lib/mix/tasks/deps.compile.ex:111: anonymous fn/2 in Mix.Tasks.Deps.Compile.do_mix/1\n    (mix) lib/mix/project.ex:196: Mix.Project.in_project/4\n    (elixir) lib/file.ex:1061: File.cd!/2\n    (mix) lib/mix/dep.ex:137: Mix.Dep.in_dependency/3\n    (mix) lib/mix/tasks/deps.compile.ex:57: anonymous fn/3 in Mix.Tasks.Deps.Compile.compile/1\n    (elixir) lib/enum.ex:977: anonymous fn/3 in Enum.map/2\n    (elixir) lib/enum.ex:1261: Enum.\"-reduce/3-lists^foldl/2-0-\"/3\n", 10])
    (elixir) lib/kernel/cli.ex:83: anonymous fn/3 in Kernel.CLI.exec_fun/2
```

**Comment by stavro** [comment]
Another example from SemaphoreCI failing to compile a stock Phoenix ErrorView:  :-/

```
== Compilation error on file web/views/error_view.ex ==

23:16:17.273 [error] GenServer :standard_error_sup terminating
Last message: {:EXIT, #PID<0.22.0>, :eagain}
State: {:state, :standard_error, :undefined, #PID<0.22.0>, {:local, :standard_error_sup}}
** (exit) :eagain
** (ArgumentError) argument error
    (stdlib) :io.put_chars(:standard_error, :unicode, [['web/views/error_view.ex', 58, '12', 58, 32], 'warning: ', [118, 97, 114, 105, 97, 98, 108, 101, 32, 'reason', 32, 105, 115, 32, 117, 110, 117, 115, 101, 100], 10])
    (elixir) src/elixir_errors.erl:255: :elixir_errors.do_warn/1
    (stdlib) lists.erl:1337: :lists.foreach/2
    (stdlib) erl_eval.erl:669: :erl_eval.do_apply/6
```

**Comment by josevalim** [comment]
@stavro it is a bug in Erlang 18. The best fix is to use Erlang 17 for now.

**Comment by josevalim** [comment]
Fixed in Erlang 18.0.3 which is just out!
